internal/provider/resource_google_hosted_conn.go

```diff
@@ -209,36 +362,75 @@ func resourceGoogeReqHostConnDelete(ctx context.Context, d *schema.ResourceData,
 }
 
 func extractGoogleReqConn(d *schema.ResourceData) packetfabric.GoogleReqHostedConn {
-	googleHosted := packetfabric.GoogleReqHostedConn{}
+	hostedGoogleConn := packetfabric.GoogleReqHostedConn{}
 	if accountUUID, ok := d.GetOk("account_uuid"); ok {
-		googleHosted.AccountUUID = accountUUID.(string)
+		hostedGoogleConn.AccountUUID = accountUUID.(string)
 	}
 	if pairingKey, ok := d.GetOk("google_pairing_key"); ok {
-		googleHosted.GooglePairingKey = pairingKey.(string)
+		hostedGoogleConn.GooglePairingKey = pairingKey.(string)
 	}
 	if vlanAttach, ok := d.GetOk("google_vlan_attachment_name"); ok {
-		googleHosted.GoogleVlanAttachmentName = vlanAttach.(string)
+		hostedGoogleConn.GoogleVlanAttachmentName = vlanAttach.(string)
 	}
 	if description, ok := d.GetOk("description"); ok {
-		googleHosted.Description = description.(string)
+		hostedGoogleConn.Description = description.(string)
 	}
 	if pop, ok := d.GetOk("pop"); ok {
-		googleHosted.Pop = pop.(string)
+		hostedGoogleConn.Pop = pop.(string)
 	}
 	if port, ok := d.GetOk("port"); ok {
-		googleHosted.Port = port.(string)
+		hostedGoogleConn.Port = port.(string)
 	}
 	if vlan, ok := d.GetOk("vlan"); ok {
-		googleHosted.Vlan = vlan.(int)
+		hostedGoogleConn.Vlan = vlan.(int)
 	}
 	if srcSvlan, ok := d.GetOk("src_svlan"); ok {
-		googleHosted.SrcSvlan = srcSvlan.(int)
+		hostedGoogleConn.SrcSvlan = srcSvlan.(int)
 	}
 	if speed, ok := d.GetOk("speed"); ok {
-		googleHosted.Speed = speed.(string)
+		hostedGoogleConn.Speed = speed.(string)
 	}
 	if poNumber, ok := d.GetOk("po_number"); ok {
-		googleHosted.PONumber = poNumber.(string)
+		hostedGoogleConn.PONumber = poNumber.(string)
+	}
+	if cloudSettingsList, ok := d.GetOk("cloud_settings"); ok {
+		cs := cloudSettingsList.([]interface{})[0].(map[string]interface{})
+		hostedGoogleConn.CloudSettings = &packetfabric.CloudSettingsHosted{}
+		hostedGoogleConn.CloudSettings.CredentialsUUID = cs["credentials_uuid"].(string)
+		hostedGoogleConn.CloudSettings.GoogleRegion = cs["google_region"].(string)
+		if googleProjectID, ok := cs["google_project_id"]; ok {
+			hostedGoogleConn.CloudSettings.GoogleProjectID = googleProjectID.(string)
+		}
+		hostedGoogleConn.CloudSettings.GoogleVlanAttachmentName = cs["google_vlan_attachment_name"].(string)
```

Comment:
the google_pairing_key is actually not set under cloud setting (we automatically handle it in our BE) so I think we are good. I will clarify the description though to avoid confusion